core/src/main/java/org/infinispan/util/concurrent/BlockingManager.java

```diff
@@ -247,12 +248,46 @@ <V> CompletionStage<V> whenCompleteBlocking(CompletionStage<V> stage,
 
    /**
     * Provides a {@link BlockingExecutor} which is limited to the provided concurrency amount.
+    *
     * @param name name of the limited blocking executor.
     * @param concurrency maximum amount of concurrent operations to be performed via the returned executor.
     * @return a blocking executor limited in the amount of concurrent invocations.
     */
    BlockingExecutor limitedBlockingExecutor(String name, int concurrency);
 
+   /**
+    * Replacement for {@link java.util.concurrent.ScheduledExecutorService#schedule(Runnable, long, TimeUnit)}} that
+    * invokes the {@code Runnable} in a blocking thread only after the elapsed time.
+    * <p>
+    * Unlike other methods in this interface, the submitting thread does not impact this method's behavior.
+    *
+    * @param runnable blocking operation that runs some code.
+    * @param delay the time from now to delay execution
+    * @param unit the time unit of the delay parameter
+    * @param traceId an identifier that can be used to tell in a trace when an operation moves between threads.
+    * @return a stage that is completed after the runnable is done or throws an exception.
+    */
+   default ScheduledCompletableStage<Void> scheduleRunBlocking(Runnable runnable, long delay, TimeUnit unit, Object traceId) {
```

Comment:
Is there a way to return the `ScheduledFuture` instead?

Reply:
I tried to do that here => https://github.com/infinispan/infinispan/pull/10606/commits/5c510feea2dd2022b0e829ad27d3eb057146217c
Is it safe to expose it?

Reply:
I mentioned it here https://github.com/infinispan/infinispan/pull/10606#discussion_r1089058467. But I would say lets keep this commit as is and you can change it in your PR so we know what it was changed for specifically.